pkg/addr/host_spec.gobra

```diff
@@ -19,49 +19,61 @@ package addr
 import (
 	"net"
 
-	. "github.com/scionproto/scion/verification/utils/definitions"
 	"github.com/scionproto/scion/verification/utils/slices"
 )
 
 pred (h HostNone) Mem() { len(h) == HostLenNone }
-pred (h HostNone) LowMem() { true }
-
-HostNone implements HostAddr {
-	(h HostNone) String() (str string) {
-		unfold acc(h.Mem(), R13/2)
-		str = h.String()
-		fold acc(h.Mem(), R13)
-	}
-}
+
+ghost
+requires h.Mem()
+decreases
+pure func (h HostNone) IsLow() bool
+
+HostNone implements HostAddr
 
 pred (h HostIPv4) Mem() {
 	len(h) == HostLenIPv4 &&
 	slices.Bytes(h, 0, len(h))
 }
-pred (h HostIPv4) LowMem() { slices.Bytes(h, 0, len(h)) }
+
+ghost
+requires h.Mem()
+decreases
+pure func (h HostIPv4) IsLow() bool
 
 HostIPv4 implements HostAddr
 
 pred (h HostIPv6) Mem() {
 	len(h) == HostLenIPv6 &&
 	slices.Bytes(h, 0, len(h))
 }
-pred (h HostIPv6) LowMem() { slices.Bytes(h, 0, len(h)) }
+
+ghost
+requires h.Mem()
+decreases
+pure func (h HostIPv6) IsLow() bool
 
 HostIPv6 implements HostAddr
 
 pred (h HostSVC) Mem() { true }
-pred (h HostSVC) LowMem() { low(h) }
 
-HostSVC implements HostAddr {
-	(h HostSVC) String() (str string) {
-		str = h.String()
-		fold acc(h.Mem(), R13)
-	}
-}
+ghost
+requires h.Mem()
+decreases
+pure func (h HostSVC) IsLow() bool
+
+// "Reveal the body" of `h.IsLow` (necessary bc. we can't implement `h.IsLow` yet).
+// TODO: Once Gobra issue #846 is resolved, remove this.
```

Comment:
I would not actually remove these functions, I would just add an implementation once 846 is resolved. please add a `trusted` annotation to these functions